compiler/testData/diagnostics/tests/inference/expectedTypeFromCast.fir.kt

```diff
@@ -17,5 +17,5 @@ val asStarList = foo() as List<*>
 
 val safeAs = foo() as? String
 
-val fromIs = foo() is String
-val fromNoIs = foo() !is String
+val fromIs = <!USELESS_IS_CHECK!>foo() is String<!>
+val fromNoIs = <!USELESS_IS_CHECK!>foo() !is String<!>
```

Comment:
Quite similar here. `foo()` returns `Nothing` (or technically, `throw`), which makes `is String` up cast. :\